scipy/stats/stats.py

```diff
@@ -3521,7 +3521,10 @@ def kendalltau(x, y, initial_lexsort=None, nan_policy='propagate'):
 
     def count_rank_tie(ranks):
         cnt = np.bincount(ranks).astype('int64', copy=False)
-        return (cnt * (cnt - 1) // 2).sum()
+        cnt = cnt[cnt > 1]
+        return ((cnt * (cnt - 1) // 2).sum(),
+            (cnt * (cnt - 1.) * (cnt - 2)).sum(),
+            (cnt * (cnt - 1.) * (2*cnt + 5)).sum())
```

Comment:
True, but in my test it doesn't appear to be much faster, as the O(n log n) cost is elsewhere, and the multiplications happen only for block of ties. Moreover, it would mean having yet another vector allocated, as we need both cnt and cnt *(cnt - 1). This could be a problem with large vectors.